docs/fundamentals/code-analysis/style-rules/formatting-rules.md

```diff
@@ -0,0 +1,1221 @@
+---
+title: Code style formatting rules
+ms.date: 09/25/2020
+ms.topic: reference
+author: gewarren
+ms.author: gewarren
+dev_langs:
+- CSharp
+- VB
+helpviewer_keywords:
+- formatting code style rules [EditorConfig]
+- formatting rules
+- EditorConfig formatting conventions
+---
+# Formatting rules
+
+Formatting rules affect how the indentation, spaces and new lines are aligned around .NET programming language constructs. They fall into the below categories:
+
+- [.NET formatting rules](#net-formatting-rules): Rules that apply to both C# and Visual Basic. The EditorConfig option names for these rules start with `dotnet_` prefix.
+- [C# formatting rules](#c-formatting-rules): Rules that are specific to C# language only. The EditorConfig option names for these rules start with `csharp_` prefix.
+
+## Rule ID: "IDE0055" (Fix formatting)
+
+All formatting options have rule ID `IDE0055` and title `Fix formatting`. Severity of formatting violations can be configured in an EditorConfig file with the following option:
+
+```ini
+dotnet_diagnostic.IDE0055.severity = <severity value>
+```
+
+The value must be `warning` or `error` to be [enforced on build](../overview.md#code-style-analysis). For all possible severity values, see [severity level](../configuration-options.md#severity-level).
+
+## Option format
+
+Options for formatting rules can be specified in an EditorConfig file with the following format:
+
+`rule_name = value`
+
+For many rules, you specify either `true` (prefer this style) or `false` (do not prefer this style) for `value`. For other rules, you specify a value such as `flush_left` or `before_and_after` to describe when and where to apply the rule. You don't specify a severity.
+
+## .NET formatting rules
```

Comment:
The individual rule/option sections starting from here will be broken into separate dedicated pages in separate follow-up PRs and populated in the code style rule index page.